livekit-plugins/livekit-plugins-asyncai/livekit/plugins/asyncai/tts.py

```diff
@@ -0,0 +1,293 @@
+    ) -> SynthesizeStream:
+        return SynthesizeStream(tts=self, conn_options=conn_options)
+
+    def synthesize(
+        self, text: str, *, conn_options: APIConnectOptions = DEFAULT_API_CONNECT_OPTIONS
+    ):
+        pass
+
+    async def aclose(self) -> None:
+        for stream in list(self._streams):
+            await stream.aclose()
+
+        self._streams.clear()
+        await self._pool.aclose()
+
+
+class SynthesizeStream(tts.SynthesizeStream):
+    def __init__(self, *, tts: TTS, conn_options: APIConnectOptions):
+        super().__init__(tts=tts, conn_options=conn_options)
+        self._tts: TTS = tts
+        self._sent_tokenizer_stream = tokenize.basic.SentenceTokenizer(
+            min_sentence_len=BUFFERED_WORDS_COUNT
+        ).stream()
+        self._opts = replace(tts._opts)
+
+    async def _run(self, output_emitter: tts.AudioEmitter) -> None:
+        request_id = utils.shortuuid()
+        output_emitter.initialize(
+            request_id=request_id,
+            sample_rate=self._opts.sample_rate,
+            num_channels=1,
+            mime_type="audio/pcm",
+            stream=True,
+        )
+
+        async def _sentence_stream_task(ws: aiohttp.ClientWebSocketResponse) -> None:
+            async for ev in self._sent_tokenizer_stream:
+                token_pkt = {}
+                token_pkt["transcript"] = ev.token + " "
+                token_pkt["force"] = True
+                self._mark_started()
+                await ws.send_str(json.dumps(token_pkt))
+
+            # end_pkt = {}
+            # end_pkt["transcript"] = ""
+            # await ws.send_str(json.dumps(end_pkt))
+
+        async def _input_task() -> None:
+            async for data in self._input_ch:
+                if isinstance(data, self._FlushSentinel):
+                    self._sent_tokenizer_stream.flush()
+                    continue
+
+                self._sent_tokenizer_stream.push_text(data)
+
+            self._sent_tokenizer_stream.end_input()
+
+        async def _recv_task(ws: aiohttp.ClientWebSocketResponse) -> None:
+            current_segment_id: str | None = None
+            while True:
+                msg = await ws.receive()
+                if msg.type in (
+                    aiohttp.WSMsgType.CLOSED,
+                    aiohttp.WSMsgType.CLOSE,
+                    aiohttp.WSMsgType.CLOSING,
+                ):
+                    raise APIStatusError(
+                        "Async connection closed unexpectedly", request_id=request_id
+                    )
+
+                if msg.type != aiohttp.WSMsgType.TEXT:
+                    logger.warning("unexpected Async message type %s", msg.type)
+                    continue
+
+                data = json.loads(msg.data)
+                if current_segment_id is None:
+                    current_segment_id = "new_segment"
+                    output_emitter.start_segment(segment_id="new_segment")
```

Comment:
It may be better to keep track of the current segment id as `self._current_segment_id`
```suggestion
                if current_segment_id is None:
                    current_segment_id = utils.shortuuid()
                    output_emitter.start_segment(segment_id=current_segment_id)
```